### ISSUE TITLE
nvim_win_get_position not work after nvim_open_win

### ISSUE THREAD

**Issue by chemzqm** [neovim/neovim#13403]
- `nvim --version`: v0.5.0-822-g4e6f00dd2
- `vim -u DEFAULTS` (version: ) behaves differently? N/A
- Operating system/version: MacOS
- Terminal name/version: iTerm2

### Steps to reproduce using `nvim -u NORC`

Source vim code: 

``` vim
let buf = nvim_create_buf(1,1)
call nvim_buf_set_lines(s:buf, 0, -1, v:false, ['a', 'b'])
let win = nvim_open_win(s:buf, v:false, {
      \ 'height':2,
      \ 'width': 10,
      \ 'col': 10,
      \ 'row': 10,
      \ 'relative': 'editor',
      \ 'style': 'minimal',
      \})
let pos = nvim_win_get_position(win)
echo pos
```

### Actual behaviour

Returned position is `[0, 0]`.

### Expected behaviour

Return window position.

I can use `redraw` to make `nvim_win_get_position` return correct window position, but seems that `redraw` should not be needed.

**Comment by folke** [comment]
Related to https://github.com/neovim/neovim/issues/14735

**Comment by dundargoc** [comment]
Get the following error opening the code snippet with `nvim --clean -u minimal.vim`:

```
Error detected while processing /home/dgoc/programs/minimal.vim:
line    2:
E121: Undefined variable: s:buf
E116: Invalid arguments for function nvim_buf_set_lines
line    3:
E121: Undefined variable: s:buf
E116: Invalid arguments for function nvim_open_win
line   11:
E121: Undefined variable: win
E116: Invalid arguments for function nvim_win_get_position
line   12:
E121: Undefined variable: pos
```

Not sure if `s:buf` is an old way to refer to current buffer or if I'm missing something.

**Comment by zeertzjq** [comment]
Seems already fixed.

**Comment by zeertzjq** [comment]
Seems already fixed.

**Comment by zeertzjq** [comment]
This was fixed by #14770.
